packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -434,9 +434,30 @@ export class StateMachine extends StateMachineBase {
     this.stateMachineArn = this.getResourceArnAttribute(resource.ref, {
       service: 'states',
       resource: 'stateMachine',
-      resourceName: this.physicalName,
+      resourceName: this.stateMachineName,
```

Comment:
Are we sure about this change? Has this bug been reported anywhere?

Reply:
Reverted.  I think that was a relic from some trial and error.